### ISSUE TITLE
Completion modal cannot be closed using Escape key

### ISSUE THREAD

**Issue by huyenltnguyen** [freeCodeCamp/freeCodeCamp#55423]
## Description

Modals in /learn are built upon the @freecodecamp/ui's `Modal` component, and by default, this `Modal` component can be closed using the Escape key. The completion modal also uses the `Modal` component as the base, but it doesn't seem to listen to the Escape keypress and remains open when the key is pressed.

I'm not sure, but I think this issue has something to do with the `handleKeypress` function:

https://github.com/freeCodeCamp/freeCodeCamp/blob/19aeb12699a1a9bf4689827fb649be11a1e925c5/client/src/templates/Challenges/components/completion-modal.tsx#L130-L138

## Affected page

The modal is used on multiple pages, but I noticed the issue when I was testing an English challenge and a legacy JS challenge.
- English challenge: https://www.freecodecamp.org/learn/a2-english-for-developers/learn-greetings-in-your-first-day-at-the-office/task-14
- Legacy JS challenge: https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/basic-javascript/comment-your-javascript-code

## Steps to reproduce

- Go to one of the pages above and complete the challenge
- When the completion modal shows up, press Escape

## Changes needed

- Fix the issue
- Add a test case to the following file to lock down the behavior:
  - https://github.com/freeCodeCamp/freeCodeCamp/blob/main/e2e/completion-modal.spec.ts

**Comment by gagan-bhullar-tech** [comment]
@huyenltnguyen The modal is closing on pressing Escape key. Checked in Chrome/MacOS. Can you please specify if this issue is for a specific browser.

**Comment by anishlukk123** [comment]
@gagan-bhullar-tech  I have a mac and I am trying to  replicate the issue here, and I enter fn + esc then it doesnt close, however it does say  that is not the right answer, is that  you mean by lockdown, second the  code mentions  cltrl +enter , but you want to hit  escape  to close the module.  can you help understand what you mean closing the module

**Comment by anishlukk123** [comment]
also  @huyenltnguyen  could  we assume that other modules do work this away ?

**Comment by gikf** [comment]
I found that being logged in makes the difference.
- When logged in {esc} closes modal
- When logged out {esc} doesn't close modal

**Comment by anishlukk123** [comment]
I have tried it logged out and logged in , and in my dev  and it doesnt close

**Comment by gagan-bhullar-tech** [comment]
@gikf for me, it is closing the model in both logged out and logged in scenarios.

**Comment by anishlukk123** [comment]
are in gitpod  gagan,  or is your own local

**Comment by gagan-bhullar-tech** [comment]
its in my own local. Checking in gitpod.

**Comment by a2937** [comment]
I don't have playwright/.auth/certified-user.json and it might make writing tests that involved a signed in user difficult.
